### PR TITLE
Add pika build for node and web

### PR DESCRIPTION
**Pull request by bendtherules**
Overall changes:
1. Add pika with node and browser build. These are plain builds, not bundles.
2. Handle json import with babel json inline plugin (pika doesn't handle this). Use `.json` in import path for this to work.
3. Changes source folder from `lib` to `src`. Fixed this paths in other places.

I have made a consumer repo to test it in browser (https://github.com/bendtherules/test-octokit-throttling). It is working there now.
Can be manually checked by doing npm link and throttling connections in devtools to high latency.

Edit: Added other changes following https://github.com/octokit/plugin-throttling.js/issues/84#issuecomment-497789779

### CODE REVIEW

**Comment by gr2m** [comment]
See my comment at https://github.com/octokit/plugin-throttling.js/issues/84#issuecomment-497746944.

- Replace the *.json file with a *.js file so we can remove the `.babelrc` file.
- Add pika bundle step for the web
- Add pika unpkg step to load the web build directly from unpkg.com
- update the README as described in https://gist.github.com/gr2m/5dd2ec6e3a04c3dc79f0cbb46decbe46
- update .travis.yml to add a build test, bundle size test and update the release step

File: .travis.yml [hunk 1]
```diff
@@ -30,7 +30,10 @@ jobs:
       node_js: lts/*
       script:
         - git checkout $TRAVIS_BRANCH
+        # generate triggers-notification-paths.js file (without proper formatting)
         - node scripts/generate-routes
+        # run standard fix on that file to fix formatting
+        - npx standard src/triggers-notification-paths.js --fix
```
Comment (gr2m) [inline .travis.yml]: smart 👍

File: .travis.yml [hunk 2]
```diff
@@ -12,17 +12,31 @@ jobs:
...
       script: npm run test:memory
+    - node_js: lts/*
+      env: build test
+      script: npm run build
+    - node_js: lts/*
+      env: bundle size
+      script:
+        - npm run build
+        - npx terser pkg/dist-web/index.bundled.js --output pkg/dist-web/index.bundled.min.js --compress --mangle
+        - npx bundlesize --files pkg/dist-web/index.bundled.min.js --max-size 3KB
```
Comment (gr2m) [inline .travis.yml]: check if the `3KB` is the right size

Comment (bendtherules) [reply .travis.yml]: made it 4KB, currently at 3.53KB

**Comment by bendtherules** [comment]
Please check if the readme example looks correct, uses proper naming, etc.

File: README.md [hunk 3]
```diff
@@ -11,6 +11,35 @@ Implements all [recommended best practises](https://developer.github.com/v3/guid
...
+Install with <code>npm install @octokit/plugin-throttling</code>
+
+```js
+const pluginThrottling = require("@octokit/plugin-throttling");
+// or: import pluginThrottling from "@octokit/plugin-throttling";
+```
+
+</td></tr>
+</tbody>
+</table>
```
Comment (gr2m) [inline README.md]: That looks good!

Once we have `@octokit/rest` available on unpkg.com we should update these examples to also load `@octokit/rest` from there. I’ll make a follow up issue :)

**Comment by gr2m** [comment]
Are you all set from your side?

**Comment by bendtherules** [comment]
> Are you all set from your side?

Yes, done from my side

**Comment by gr2m** [comment]
One problem is that the web bundling is not working, because the source is not use ES modules. 

I think you should be able to replace the `require` calls with `import`s.

You can also change the default export to a named export, as I’ve found this to be most future proof during the transitioning time between ES modules and commonjs modules:

```js
const { throttlingPlugin } = require('@octokit/plugin-throttling')
```

Would you like to give that a try? We can also do that in a follow up PR if the current PR resolves #84 for you

**Comment by bendtherules** [comment]
> One problem is that the web bundling is not working, because the source is not use ES modules.
> 
> I think you should be able to replace the `require` calls with `import`s.

That means, converting the source files to always use esm syntax for import/export? 
 
> You can also change the default export to a named export, as I’ve found this to be most future proof during the transitioning time between ES modules and commonjs modules:
> 
> ```js
> const { throttlingPlugin } = require('@octokit/plugin-throttling')
> ```
I tried doing this, but it doesn't seem to help. The require calls are still not being replaced in web build and bundle. Can you show an example of this?




> Would you like to give that a try? We can also do that in a follow up PR if the current PR resolves #84 for you

**Comment by gr2m** [comment]
> That means, converting the source files to always use esm syntax for import/export? This sounds better.

Yes, exactly.

> Can you show an example of this?

https://github.com/octokit/endpoint.js/blob/f44b59151d601dc6c5fc600880c73fd5e20ade3c/src/index.ts#L4

So I’d do a 

```js
export const pluginThrottling = ...
```

In this case
```

**Comment by bendtherules** [comment]
Screenshot 2019-06-01 at 4 47 32 AM

Doing this still outputs this bundle

Screenshot 2019-06-01 at 4 48 25 AM

What am i doing wrong here?

**Comment by gr2m** [comment]
You also need to replace all `require` statements in `src/` with ES `import` statements

**Comment by bendtherules** [comment]
> You also need to replace all `require` statements in `src/` with ES `import` statements

Ok, so effectively pika doesn't seem to support commonjs export/exports?

Anyway, i'll change them to esm syntax and see.

**Comment by gr2m** [comment]
> Ok, so effectively pika doesn't seem to support commonjs export/exports

Yes. The main focus for Pika is to create more packages with a "module" export that can directly be used in browsers. It created a search for a subset of npm’s packages that export a module:
https://www.pikapkg.com/

**Comment by gr2m** [comment]
Hey Abhas, do you still plan to finish this up? No rush, just checking in right now, it’s not a blocker yet

**Comment by bendtherules** [comment]
Hey,
I'll try to pick it up this weekend. If not, please feel free to assign to someone else.

**Comment by bendtherules** [comment]
Changed the imports now.

Will have to merge master changes into it. I was facing some problem with node-memwatch node-gyp build, can't find some header files on Windows.

File: src/index.js [hunk 4]
```diff
@@ -37,7 +37,7 @@ const createGroups = function (Bottleneck, common) {
   })
 }
 
-function throttlingPlugin (octokit, octokitOptions = {}) {
+export default function throttlingPlugin (octokit, octokitOptions = {}) {
```
Comment (gr2m) [inline src/index.js]: I would prefer to `export function throttlingPlugin`, it's easier for compatibility between ES Modules and commonJS. You’ll have to update the README imports as well

```js
import { pluginThrottling } from "https://unpkg.com/@octokit/plugin-throttling";
```

Sorry that I only found out about this now

**Comment by bendtherules** [comment]
TODO: Fix pretest scripts. Right now, ignores `standard` lint errors and isn't organised well.

**Comment by bendtherules** [comment]
> ```json
>     "before:test": "npm run -s build",
>     "test": "nyc mocha test/integration/",
>     "before:test:memory": "npm run -s build",
>     "test:memory": "node test/memory-leakage-test"
> ```

before:test doesn't seem to work with npm scripts. Only pretest is allowed, that too for just `test` and not test variants (as they are not recognised as such).

Probably should move them to bash files.

**Comment by bendtherules** [comment]
@gr2m Named export and test scripts changes are done.

Comment on coverage: https://github.com/octokit/plugin-throttling.js/pull/89#issuecomment-507253782

**Comment by gr2m** [comment]
I'm very sorry Abhas, I dropped the ball on this. I had to go ahead and migrate the repository to TypeScript, and I've setup pika in the first step, in #125

**Pull request closed without merging**